sdv/evaluation/__init__.py

```diff
@@ -78,67 +94,164 @@ def get_descriptor_values(real, synth, descriptor):
 
     Args:
-        real (dict[str, pandas.DataFrame] or pandas.DataFrame):
-            Map of names and tables of real data.
         synth (dict[str, pandas.DataFrame] or pandas.DataFrame):
             Map of names and tables of synthesized data.
+        real (dict[str, pandas.DataFrame] or pandas.DataFrame):
+            Map of names and tables of real data. Defaults to ``None``.
+        metadata (str, dict or Metadata):
+            String or dictionary to instance a Metadata object or a Metadata itself.
+            Defaults to ``None``.
         descriptors (list[callable]):
             List of descriptors.
         metrics (list[callable]):
             List of metrics.
+        root_path (str):
+            Relative path to find the metadata.json file when needed. Defaults to ``None``.
+        table_name (str):
+            Table name to be evaluated, only used when ``synth`` is a ``pandas.DataFrame``
+            and ``real`` is ``None``. Defaults to None.
+        by_tables (bool):
+            Flag to return a ``pandas.DataFrame`` when ``True``, ``pandas.Series`` otherwise.
 
     Return:
-        pandas.Series:
+        pandas.Series or pandas.DataFrame:
             It has the metrics as index, and the scores as values.
     """
-    if isinstance(real, pd.DataFrame):
-        described = get_descriptors_table(real, synth, descriptors)
+    synth, real, metadata = _validate_arguments(synth, real, metadata, root_path, table_name)
 
-    if isinstance(real, dict):
-        assert real.keys() == synth.keys(), "real and synthetic dataset must have the same tables"
-        result = list()
+    results = dict()
+    for name, real_table in real.items():
+        synth_table = synth[name]
+        table_scores = get_descriptors_table(
+            real_table, synth_table, metadata, name, descriptors)
+        results[name] = table_scores
 
-        for name, real_table in real.items():
-            synth_table = synth[name]
-            result.append(get_descriptors_table(real_table, synth_table, descriptors))
+    if not by_tables:
+        results = {0: pd.concat(list(results.values()), axis=1)}
 
-        described = pd.concat(result, axis=1).fillna(0)
+    if not isinstance(metrics, (list, tuple)):
+        metrics = [metrics]
 
-    real_descriptors = described.iloc[0, :]
-    synth_descriptors = described.iloc[1, :]
     scores = dict()
+    for name, described in results.items():
+        table_scores = dict()
+        real_descriptors = described.iloc[0, :]
+        synth_descriptors = described.iloc[1, :]
+        for metric in metrics:
+            table_scores[metric.__name__] = metric(real_descriptors, synth_descriptors)
+
+        scores[name] = table_scores
+
+    scores = pd.DataFrame(scores).T
 
-    for metric in metrics:
-        scores[metric.__name__] = metric(real_descriptors, synth_descriptors)
+    if len(scores) > 1:
+        mean = scores.mean()
+        mean.name = '__mean__'
```

Comment:
Let's use `'mean'` instead of `__mean__`.
If the name `mean` already exists in the DF, this will duplicate it, but it's OK.